examples/load_balancer/load_balancer.c

```diff
@@ -240,49 +251,83 @@ parse_app_args(int argc, char *argv[], const char *progname) {
 /*
  * This function parses the backend config. It takes the filename
  * and fills up the backend_server array. This includes the mac and ip
- * address of the backend servers
+ * address of the backend servers as well as their weights
  */
 static int
-parse_backend_config(void) {
-        int ret, temp, i;
-        char ip[32];
-        char mac[32];
-        FILE *cfg;
+parse_backend_json_config(void) {
+        int ret, i;
+        i = 0;
 
-        cfg = fopen(lb->cfg_filename, "r");
-        if (cfg == NULL) {
-                rte_exit(EXIT_FAILURE, "Error openning server \'%s\' config\n", lb->cfg_filename);
+        cJSON *config_json = onvm_config_parse_file(lb->cfg_filename);
+        cJSON *list_size = NULL;
+        cJSON *policy = NULL;
+        cJSON *ip_addr = NULL;
+        cJSON *mac_addr = NULL;
+        cJSON *weight = NULL;
+
+        if (config_json == NULL) {
+                rte_exit(EXIT_FAILURE, "%s file could not be parsed/not found. Assure config file"
+                                       " the directory to the config file is being specified.\n", lb->cfg_filename);
         }
-        ret = fscanf(cfg, "%*s %d", &temp);
-        if (temp <= 0) {
-                rte_exit(EXIT_FAILURE, "Error parsing config, need at least one server configurations\n");
+
+        config_json = config_json -> child;
+
+        list_size = cJSON_GetObjectItem(config_json, "LIST_SIZE");
+        policy = cJSON_GetObjectItem(config_json, "policy");
+        
+        if (list_size == NULL) rte_exit(EXIT_FAILURE, "LIST_SIZE not found/invalid\n");
+        if (policy == NULL) rte_exit(EXIT_FAILURE, "Policy not found/invalid\n");
+
+        lb->server_count = list_size->valueint;
+        lb->policy = strdup(policy->valuestring);
+
+        if (!((!strcmp(lb->policy,"random")) || (!strcmp(lb->policy,"rrobin")) || (!strcmp(lb->policy,"weighted_random")))) {
+                rte_exit(EXIT_FAILURE, "Invalid policy. Check server.conf\n");
         }
-        lb->server_count = temp;
+        
+        
+        lb->weights = (int*)calloc(lb->server_count,sizeof(int));
```

Comment:
only have 1 blank line between code sections. If code is closely related, don't put empty lines.